components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/SftpOperations.java

```diff
@@ -714,97 +722,104 @@ private boolean retrieveFileToStreamInBody(String name, Exchange exchange) throw
-            }
-
-            channel.get(remoteName, os);
-
-        } catch (SftpException e) {
-            LOG.trace("Error occurred during retrieving file: {} to local directory. Deleting local work file: {}", name, temp);
-            // failed to retrieve the file so we need to close streams and delete in progress file
-            // must close stream before deleting file
-            IOHelper.close(os, "retrieve: " + name, LOG);
-            boolean deleted = FileUtil.deleteFile(temp);
-            if (!deleted) {
-                LOG.warn("Error occurred during retrieving file: " + name + " to local directory. Cannot delete local work file: " + temp);
+            
+                // store content as a file in the local work directory in the temp handle
+                os = new FileOutputStream(temp);
+            
+                // set header with the path to the local work file
+                exchange.getIn().setHeader(Exchange.FILE_LOCAL_WORK_PATH, local.getPath());
+            } catch (Exception e) {
+                throw new GenericFileOperationFailedException("Cannot create new local work file: " + local);
             }
-            throw new GenericFileOperationFailedException("Cannot retrieve file: " + name, e);
-        } finally {
-            IOHelper.close(os, "retrieve: " + name, LOG);
-
-            // change back to current directory if we changed directory
-            if (currentDir != null) {
-                changeCurrentDirectory(currentDir);
+            String currentDir = null;
+            try {
+                // store the java.io.File handle as the body
+                file.setBody(local);
+            
+                String remoteName = name;
+                if (endpoint.getConfiguration().isStepwise()) {
+                    // remember current directory
+                    currentDir = getCurrentDirectory();
+            
+                    // change directory to path where the file is to be retrieved
+                    // (must do this as some FTP servers cannot retrieve using absolute path)
+                    String path = FileUtil.onlyPath(name);
+                    if (path != null) {
+                        changeCurrentDirectory(path);
+                    }
+                    // remote name is now only the file name as we just changed directory
+                    remoteName = FileUtil.stripPath(name);
+                }
+            
+                channel.get(remoteName, os);
+            
+            } catch (SftpException e) {
+                LOG.trace("Error occurred during retrieving file: {} to local directory. Deleting local work file: {}", name, temp);
+                // failed to retrieve the file so we need to close streams and delete in progress file
+                // must close stream before deleting file
+                IOHelper.close(os, "retrieve: " + name, LOG);
+                boolean deleted = FileUtil.deleteFile(temp);
```

Comment:
the boolean `deleted` is in use only once inside the conditional statement below so I think it could be omitted like line 759 and line 817:
```java
if (!FileUtil.deleteFile(temp)) {
```